### PR TITLE
Bluetooth: controller: Fix for invalid ctrl pdu len and other refactoring

### PR DESCRIPTION
**Pull request by cvinayak**
Refer to individual commits for details.

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 1]
```diff
@@ -52,22 +49,23 @@
...
 #endif /* CONFIG_BT_CTLR_CONN_RSSI */
 
-#define RADIO_IRK_COUNT_MAX	8
-
 #define SILENT_CONNECTION	0
 
+#define RADIO_TIFS                      150
+#define RADIO_CONN_EVENTS(x, y)         ((u16_t)(((x) + (y) - 1) / (y)))
+
+#define RADIO_TICKER_JITTER_US           16
```
Comment (carlescufi) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Should these be part of the hal, or are they common to all radios?

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Comment by carlescufi** [comment]
I really like your look-up table for pdu lengths, very efficient.

File: subsys/bluetooth/controller/ll_sw/ctrl.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: The value is used in a mixed up context, both in relation to BLE permitted sleep clock accuracy jitter of +/-16us and ticker's API limitation where in first interval having a jitter of half a 32KHz unit.
I will prefer to take this up at the next opportunity (true portability to other SoCs).

Comment (carlescufi) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: OK, makes sense. I have no issues with the rest of the PR.

**Review by carlescufi — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/pdu.h [hunk 2]
```diff
@@ -224,6 +225,14 @@ struct pdu_data_llctrl_enc_rsp {
 	u8_t ivs[4];
 } __packed;
 
+struct pdu_data_llctrl_start_enc_req {
+	u8_t  empty[0];
```
Comment (jhedberg) [inline subsys/bluetooth/controller/ll_sw/pdu.h]: Does the compiler really complain if you have empty structs (with no members inside them)? I'd have assumed it doesn't.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: It did not, and am glad, as this allowed me to use offsetof and sizeof values in lookup table.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: Sorry... you mean you do not want "empty[0];"  ?

Comment (jhedberg) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: yes, that's what I meant. If it's not strictly required then it seems a bit pointless.

Comment (jhedberg) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: if you want you can replace it with /* no parameters */ or something like that

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/pdu.h]: will do.

**Comment by cvinayak** [comment]
@jhedberg and @carlescufi remove the empty[0] members.

**Comment by carlescufi** [comment]
@jhedberg are you happy with this?

**Review by jhedberg — APPROVED** [review]
> @jhedberg are you happy with this?

@carlescufi yes looks fine

**Pull request merged**